pkg/spec/deploymentconfig.go

```diff
@@ -0,0 +1,165 @@
+	}
+
+	// fix deployment.VolumeClaims
+	deployment.VolumeClaims, err = fixVolumeClaims(deployment.VolumeClaims, deployment.Name)
+	if err != nil {
+		return errors.Wrap(err, "Unable to fix persistentVolume")
+	}
+
+	// fix deployment.configMaps
+	deployment.ConfigMaps, err = fixConfigMaps(deployment.ConfigMaps, deployment.Name)
+	if err != nil {
+		return errors.Wrap(err, "unable to fix configMaps")
+	}
+
+	deployment.Containers, err = fixContainers(deployment.Containers, deployment.Name)
+	if err != nil {
+		return errors.Wrap(err, "unable to fix containers")
+	}
+
+	deployment.InitContainers, err = fixContainers(deployment.InitContainers, deployment.Name)
+	if err != nil {
+		return errors.Wrap(err, "unable to fix init-containers")
+	}
+
+	deployment.Secrets, err = fixSecrets(deployment.Secrets, deployment.Name)
+	if err != nil {
+		return errors.Wrap(err, "unable to fix secrets")
+	}
+
+	return nil
+}
+
+func (deployment *DeploymentConfigSpecMod) Transform() ([]runtime.Object, []string, error) {
+
+	// Create Kubernetes objects (since OpenShift uses Kubernetes underneath, no need to refactor
+	// this portion
+	runtimeObjects, extraResources, err := deployment.CreateK8sObjects()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to create Kubernetes objects")
+	}
+
+	// Set appropriate GVK BEFORE adding DeploymentConfig controller
+	// as OpenShift controllers are not available in the Kubernetes controller / setGVK check
+	scheme, err := GetScheme()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "unable to get scheme")
+	}
+
+	// Set's the appropriate GVK
+	for _, runtimeObject := range runtimeObjects {
+		if err := SetGVK(runtimeObject, scheme); err != nil {
+			return nil, nil, errors.Wrap(err, "unable to set Group, Version and Kind for generated Kubernetes resources")
+		}
+	}
+
+	// Create the DeploymentConfig controller!
+	deploy, err := deployment.createDeploymentConfigController()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to create Kubernetes Deployment controller")
+	}
+
+	// adding controller objects
+	// deployment will be nil if no deployment is generated and no error occurs,
+	// so we only need to append this when a legit deployment resource is returned
+	if deploy != nil {
+		runtimeObjects = append(runtimeObjects, deploy)
+		log.Debugf("deployment: %s, deployment: %s\n", deploy.Name, spew.Sprint(deployment))
+	}
+
+	if len(runtimeObjects) == 0 {
+		return nil, nil, errors.New("No runtime objects created, possibly because not enough input data was passed")
+	}
+
+	return runtimeObjects, extraResources, nil
+}
+
+// Created the OpenShift DeploymentConfig controller
+func (deployment *DeploymentConfigSpecMod) createDeploymentConfigController() (*os_deploy_v1.DeploymentConfig, error) {
+
+	// Set replicas to 1 if not set (0)
+	replicas := deployment.Replicas
+	if replicas == 0 {
+		replicas = 1
+	}
+
+	return &os_deploy_v1.DeploymentConfig{
+		TypeMeta: metav1.TypeMeta{
```

Comment:
this is created on the fly, you don't need to hard code the type meta!

Reply:
Actually, we do. SetGVK and such as implemented by @containscafeine only supported Kubernetes, thus we have to manually specify the TypeMeta.

Reply:
Whoops, we need to fix this in another PR then, need to check for OpenShift does this!

Reply:
Tracked in #349